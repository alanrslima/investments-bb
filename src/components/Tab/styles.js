import { StyleSheet } from 'react-native';
import { colors, fonts } from '../../styles';

export default StyleSheet.create({
  container: {
    backgroundColor: colors.WHITE,
  },
  content: {
    marginLeft: 20,
    paddingVertical: 15,
    borderBottomColor: colors.GRAY,
    borderBottomWidth: 2,
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingRight: 20,
  },
  title: {
    color: colors.BLACK,
    fontWeight: '600',
    fontSize: fonts.TITLE,
  },
  description: {
    color: colors.GRAY_DARK,
    fontWeight: '600',
    fontSize: fonts.TITLE,
  }
});
