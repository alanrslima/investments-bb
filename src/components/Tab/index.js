import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

function Tab({
  title, description, containerStyle, error
}) {
  return (
    <View style={[styles.container, containerStyle]}>
      <View style={styles.content}>
        <View style={styles.contentInfo}>
          <Text style={styles.title}>{title}</Text>
          <Text style={styles.description}>{description}</Text>
        </View>
        {error ? <Text style={styles.textError}>{error}</Text> : null}
      </View>
    </View>
  );
}

Tab.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  containerStyle: PropTypes.objectOf(PropTypes.number),
  error: PropTypes.string,
};

Tab.defaultProps = {
  containerStyle: {},
  error: '',
};

export default Tab;
