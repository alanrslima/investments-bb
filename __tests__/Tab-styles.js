import React from 'react';
import renderer from 'react-test-renderer';
import Tab from '../src/components/Tab';


test('renders correctly', () => {
  const tree = renderer.create(<Tab
    description="Investimento I"
    title="Nome"
    error="Nome invÃ¡lido"
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
