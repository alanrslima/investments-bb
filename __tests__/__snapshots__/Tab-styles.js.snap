// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#F4F4F4",
        "borderBottomWidth": 2,
        "marginLeft": 20,
        "paddingRight": 20,
        "paddingVertical": 15,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Nome
      </Text>
      <Text
        style={
          Object {
            "color": "#707070",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Investimento I
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#FF0000",
          "fontSize": 12,
          "marginTop": 5,
        }
      }
    >
      Nome inv√°lido
    </Text>
  </View>
</View>
`;
